### DISCUSSION TITLE
1 controller, 2 machines?

### DISCUSSION THREAD

**Discussion by frdfsnlght** [terjeio/grblHAL#168] [Q&A]
I'm in the process of building a router, but I also plan on converting a mini lathe to CNC soon. I just had the idea of building a single controller and using it for both machines. I don't plan on using the machines simultaneously. I had planned on using aircraft connectors for all the connections, so physically connecting the controller to each machine is not a problem. But I was wondering if a single firmware build on the controller might be able to handle both machines, with just a bunch of grblHAL setting changes. I plan on using the @phil-barrett [Teensy 4.1 board](https://github.com/phil-barrett/grblHAL-teensy-4.x) (hopefully the newer version still in the works).

The main mechanical differences I see between the machines are:

* 4 vs 2 (or maybe 3) steppers
* number of limit switches
* lathe will have spindle encoder/index signals

Obviously things like steps/mm would be different for the axis, but that's a grbl setting. I think some of the limit switch stuff is also grbl settings. I imagine a "gcode" file containing the grbl setting commands for each machine. Once the controller is connected to a machine, I just run the appropriate file to set the controller to the right "mode" and off I go.

Is this possible? This could save me some serious money and space in my shop.

**Comment by terjeio** [comment]
> Is this possible?

As long as there are no compile time configurations that are different then this should work.
